README.md

```diff
@@ -1,407 +1,436 @@
+nnoremap <leader>ff <cmd>Telescope find_files<cr>
+nnoremap <leader>fg <cmd>Telescope live_grep<cr>
+nnoremap <leader>fb <cmd>Telescope buffers<cr>
+nnoremap <leader>fh <cmd>Telescope help_tags<cr>
 
-```vim
-nnoremap <Leader>p <cmd>lua require'telescope.builtin'.git_files{}<CR>
+" Using lua functions
+nnoremap <leader>ff <cmd>lua require('telescope.builtin').find_files()<cr>
+nnoremap <leader>fg <cmd>lua require('telescope.builtin').live_grep()<cr>
+nnoremap <leader>fb <cmd>lua require('telescope.builtin').buffers()<cr>
+nnoremap <leader>ff <cmd>lua require('telescope.builtin').help_tags()<cr>
 ```
 
-Searches over files in a git folder. Note: This does not work outside a git repo folder.
+See [built-in pickers](#built-in-pickers)
+  for the list of all built-in functions.
 
-```vim
-nnoremap <Leader>p <cmd>lua require'telescope.builtin'.find_files{}<CR>
-```
 
-Search over files in your `cwd` current working directory.
+<!-- Section -->
 
-```vim
-nnoremap <silent> gr <cmd>lua require'telescope.builtin'.lsp_references{}<CR>
-```
-
-Search over variable references from your Language Server.
-
-```vim
-nnoremap <Leader>en <cmd>lua require'telescope.builtin'.find_files{ cwd = "~/.config/nvim/" }<CR>
-```
-
-Find all the files in your nvim config.
 
-### Available keys for `defaults`
+## Customization
 
-- `generic_sorter`:
-    - Description: The sorter to be used for generic searches.
-    - `default`: `require('telescope.sorters').get_generic_fuzzy_sorter
-- `file_sorter`:
-    - Description: The sorter to be used for file based searches.
-    - `default`: `require('telescope.sorters').get_fuzzy_file
+You can configure `telescope.nvim` at two levels:
+  globally, through the `setup()` method,
+  or in a finer-grained manner
+  by passing options to individual built-in pickers.
 
-### Full Example
+|  [Presentation](#presentation) | [Sorting](#sorting) | [Mappings](#mappings)
 
-```vim
-lua <<EOF
--- totally optional to use setup
-require('telescope').setup {
-  defaults = {
-    shorten_path = false -- currently the default value is true
-  }
-}
-EOF
+#### Presentation
 
-nnoremap <c-p> :lua require'telescope.builtin'.find_files{}<CR>
-nnoremap <silent> gr <cmd>lua require'telescope.builtin'.lsp_references{ shorten_path = true }<CR>
-```
+| Description                                           | Keys                   | Options                    |
+|-------------------------------------------------------|------------------------|----------------------------|
+| Where the prompt should be located.                   | `prompt_position`      | top/bottom                 |
+| What should the prompt prefix be.                     | `prompt_prefix`        | string                     |
+| Where first selection should be located.              | `sorting_strategy`     | descending/ascending       |
+| How the telescope is drawn.                           | `layout_strategy`      | center/horizontal/vertical |
```

Comment:
`flex` layout should be added here as well. Also, do you think providing a visual representation of the layout is a good idea? Like the one inside `layout_strategies.lua` which looks something like this.
```
   +-----------------+---------------------+
   |                 |                     |
   |     Results     |                     |
   |                 |       Preview       |
   |                 |                     |
   +-----------------|                     |
   |     Prompt      |                     |
   +-----------------+---------------------+
```

I think it will give a better idea for people to understand what the layout does without having to try it themself.

Reply:
@elianiva I think it does, perhaps we can add a section that defines the layouts . Can post here a drawing of the available layout so later can add them. otherwise you are welcome to contribute through my fork under rewrite readme branch in which I will later merge to here.

Reply:
I'll just post them here because I'm too lazy :satisfied:

horizontal
```
   +-----------------+---------------------+
   |                 |                     |
   |     Results     |                     |
   |                 |       Preview       |
   |                 |                     |
   +-----------------|                     |
   |     Prompt      |                     |
   +-----------------+---------------------+
```

vertical 
```
    +-----------------+
    |    Previewer    |
    |    Previewer    |
    |    Previewer    |
    +-----------------+
    |     Result      |
    |     Result      |
    |     Result      |
    +-----------------+
    |     Prompt      |
    +-----------------+
```

center
```
    +--------------+
    |    Preview   |
    +--------------+
    |    Prompt    |
    +--------------+
    |    Result    |
    |    Result    |
    |    Result    |
    +--------------+
```

as for flex, they are combination of vertical and horizontal depending on screen width so it doesn't have the drawing